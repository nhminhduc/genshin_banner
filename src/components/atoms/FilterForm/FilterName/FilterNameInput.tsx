import FilterInput from "components/atoms/FilterForm/FilterInput";
import { useFilterContext } from "hooks/useFilterContext";
import React, { useRef, useState } from "react";

const FilterNameInput = () => {
  const ref = useRef(null);
  const { filterByName, nameFilter } = useFilterContext();
  const [value, setValue] = useState(nameFilter);

  const handleOnChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    setValue(event.currentTarget.value);
    filterByName(event.currentTarget.value);
  };

  return (
    <FilterInput
      className="border-2 rounded border-amber-500 w-36 md:w-48"
      id="name"
      innerRef={ref}
      name="name"
      onChange={handleOnChange}
      placeholder="Filter by name"
      type="text"
      value={value || ""}
    />
  );
};

export default FilterNameInput;
