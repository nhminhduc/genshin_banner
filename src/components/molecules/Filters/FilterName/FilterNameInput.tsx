import FilterInput from "components/atoms/FilterForm/FilterInput";
import { useDebounce } from "hooks/useDebounce";
import { useFilterContext } from "hooks/useFilterContext";
import React, { useEffect, useRef, useState } from "react";

const FilterNameInput = () => {
  const ref = useRef(null);
  const { filterByName, nameFilter } = useFilterContext();
  const [value, setValue] = useState(nameFilter);

  const debouncedSearchName = useDebounce(value, 250);

  const handleOnChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    setValue(event.currentTarget.value);
  };

  useEffect(() => {
    filterByName(debouncedSearchName);
  }, [debouncedSearchName]);

  return (
    <FilterInput
      className="border-2 rounded border-amber-500 w-36 md:w-48"
      id="name"
      innerRef={ref}
      name="name"
      onChange={handleOnChange}
      placeholder="Filter by name"
      type="text"
      value={value || ""}
    />
  );
};

export default FilterNameInput;
