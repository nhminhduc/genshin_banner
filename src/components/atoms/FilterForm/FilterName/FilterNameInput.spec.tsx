import { render } from "@testing-library/react";
import CharacterContextProvider from "context/CharacterContext";
import FilterContextProvider from "context/FilterContext";
import { describe, expect, test } from "vitest";

import FilterNameInput from "./FilterNameInput";

describe("FilterNameInput test", () => {
  test("FilterNameInput load", () => {
    const { container } = render(
      <FilterContextProvider>
        <CharacterContextProvider>
          <FilterNameInput />
        </CharacterContextProvider>
      </FilterContextProvider>,
    );
    expect(container).toMatchSnapshot();
  });
});
