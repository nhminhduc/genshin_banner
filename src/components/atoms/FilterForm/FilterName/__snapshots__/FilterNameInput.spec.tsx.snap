// Vitest Snapshot v1

exports[`FilterNameInput test > FilterNameInput load 1`] = `
<div>
  <div>
    <input
      class="border-2 rounded border-amber-500 w-36 md:w-48"
      defaultvalue=""
      id="name"
      name="name"
      placeholder="Filter by name"
      type="text"
    />
  </div>
</div>
`;
